resources/js/common/components/GlobalSearch/components/SearchResults/SearchResults.tsx

```diff
@@ -0,0 +1,194 @@
+      limit: 3,
+      icon: LuUsers,
+
+      render: (user) => {
+        const safeUser = user as App.Data.User;
+
+        return (
+          <BaseCommandItem
+            key={`user-${safeUser.displayName}`}
+            onSelect={() => {
+              window.location.assign(`/user/${safeUser.displayName}`);
+              onClose();
+            }}
+          >
+            <UserResultDisplay user={safeUser} />
+          </BaseCommandItem>
+        );
+      },
+    },
+    {
+      key: 'games',
+      heading: t('Games'),
+      results: searchResults.results.games || [],
+      relevance: searchResults.scopeRelevance.games || 0,
+      limit: 6,
+      icon: FaGamepad,
+
+      render: (game) => (
+        <BaseCommandItem
+          key={`game-${game.id}`}
+          className="group"
+          onSelect={() => {
+            window.location.assign(`/game/${game.id}`);
+            onClose();
+          }}
+        >
+          <GameResultDisplay game={game as App.Platform.Data.Game} />
+        </BaseCommandItem>
+      ),
+    },
+    {
+      key: 'hubs',
+      heading: t('Hubs'),
+      results: searchResults.results.hubs || [],
+      relevance: searchResults.scopeRelevance.hubs || 0,
+      limit: 4,
+      icon: LuNetwork,
+
+      render: (hub) => (
+        <BaseCommandItem
+          key={`hub-${hub.id}`}
+          onSelect={() => {
+            window.location.assign(`/hub/${hub.id}`);
+            onClose();
+          }}
+        >
+          <HubResultDisplay hub={hub as App.Platform.Data.GameSet} />
+        </BaseCommandItem>
+      ),
+    },
+    {
+      key: 'achievements',
+      heading: t('Achievements'),
+      results: searchResults.results.achievements || [],
+      relevance: searchResults.scopeRelevance.achievements || 0,
+      limit: 3,
+      icon: ImTrophy,
+
+      render: (achievement) => (
+        <BaseCommandItem
+          key={`achievement-${achievement.id}`}
+          onSelect={() => {
+            window.location.assign(`/achievement/${achievement.id}`);
+            onClose();
+          }}
+        >
+          <AchievementResultDisplay achievement={achievement as App.Platform.Data.Achievement} />
+        </BaseCommandItem>
+      ),
+    },
+  ];
+
+  const sectionsWithResults = sections.filter((section) => section.results.length > 0);
+
+  // Use smart section ordering with fallback to logical defaults.
+  sectionsWithResults.sort((a, b) => {
+    // If one section has significantly higher relevance (>0.3 difference), prioritize it.
+    const relevanceDiff = b.relevance - a.relevance;
+    if (Math.abs(relevanceDiff) > 0.3) {
+      return relevanceDiff > 0 ? 1 : -1;
+    }
+
+    // Otherwise, use logical default ordering.
+    const defaultOrder: Record<string, number> = {
+      games: 1,
+      hubs: 2,
+      users: 3,
+      achievements: 4,
+    };
+
+    return (defaultOrder[a.key] as number) - (defaultOrder[b.key] as number);
+  });
```

Comment:
The grouping for "All" is inconsistent. If I search on "2025", I get Hubs, Users, Games, Achievements. Searching on "crap" returns Users, Achievements and Games. If I search on "tele", I get Games, Hubs, Users, Achievements. If it was prioritizing based on match weight, I'd expect the "tele" user to be the first result.